pkg/v3/tickers/block.go

```diff
@@ -0,0 +1,74 @@
+package tickers
+
+import (
+	"context"
+	"log"
+	"sync"
+
+	ocr2keepers "github.com/smartcontractkit/ocr2keepers/pkg"
+)
+
+type blockSubscriber interface {
+	// Subscribe provides an identifier integer, a new channel, and potentially an error
+	Subscribe() (int, chan ocr2keepers.BlockHistory, error)
+	// Unsubscribe requires an identifier integer and indicates the provided channel should be closed
+	Unsubscribe(int) error
+}
+
+// blockTicker is a struct that follows the same design paradigm as a time ticker but provides blocks
+// instead of time
+type blockTicker struct {
+	C          chan ocr2keepers.BlockHistory
+	chID       int
+	ch         chan ocr2keepers.BlockHistory
+	subscriber blockSubscriber
+	closer     sync.Once
+	ctx        context.Context
+	cancel     context.CancelFunc
```

Comment:
Do we need these? If the Start method takes the run context the constructor cannot be the context provider.